akka-actor/src/main/scala/akka/pattern/internal/BackoffOnStopSupervisor.scala

```diff
@@ -88,9 +88,9 @@ import scala.concurrent.duration.FiniteDuration
         case Some(fsm) if fsm(msg) => context.stop(self)
         case _ =>
           handlerWhileStopped match {
-            case None           => context.system.deadLetters.forward(msg)
             case Some(Left(r))  => sender() ! r
-            case Some(Right(p)) => getOrCreateHandler(p).forward(msg)
+            case Some(Right(h)) => h.forward(msg)
+            case None           => context.system.deadLetters.forward(msg)
```

Comment:
could we even remove the Option for `handlerWhileStopped`? `Either[Any, ActorRef]` and populate the Right with system.deadLetters if the custom ActorRef is not given?

Reply:
quickly looked into it, but this requires access to the system/context which is only available here.

Reply:
Would be a little simpler, but requires access to context/system (which would be a breaking change to the API to have it there).

Reply:
ok, np